src/modules/Game.class.js

```diff
@@ -6,63 +7,216 @@
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
-  }
-
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  constructor(initialState = null) {
+    this.initialValue = initialState
+      ? initialState.map((row) => [...row])
+      : null;
+
+    this.board = this.initialValue
+      ? this.initialValue.map((row) => [...row])
+      : this.createEmptyBoard();
+
+    this.score = 0;
+    this.status = 'idle';
+  }
+
+  createEmptyBoard() {
+    return Array.from({ length: 4 }, () => Array(4).fill(0));
+  }
+
+  getState() {
+    return this.board;
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  getStatus() {
+    return this.status;
+  }
+
+  generateRandomTile() {
+    const emptyCells = [];
+
+    for (let row = 0; row < 4; row++) {
+      for (let col = 0; col < 4; col++) {
+        if (this.board[row][col] === 0) {
+          emptyCells.push({ row, col });
+        }
+      }
+    }
+
+    if (emptyCells.length > 0) {
+      const { row, col } =
+        emptyCells[Math.floor(Math.random() * emptyCells.length)];
+
+      this.board[row][col] = Math.random() < 0.9 ? 2 : 4;
+    }
+  }
+
+  canMove() {
+    for (let row = 0; row < 4; row++) {
+      for (let col = 0; col < 4; col++) {
+        if (this.board[row][col] === 0) {
+          return true;
+        }
+
+        if (col < 3 && this.board[row][col] === this.board[row][col + 1]) {
+          return true;
+        }
+
+        if (row < 3 && this.board[row][col] === this.board[row + 1][col]) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  moveRowLeft(row) {
+    const filteredRow = row.filter((cell) => cell !== 0);
+
+    for (let i = 0; i < filteredRow.length - 1; i++) {
+      if (filteredRow[i] === filteredRow[i + 1]) {
+        filteredRow[i] *= 2;
+        this.score += filteredRow[i];
+        filteredRow[i + 1] = 0;
+      }
+    }
+
+    const mergedRow = filteredRow.filter((cell) => cell !== 0);
+
+    while (mergedRow.length < 4) {
+      mergedRow.push(0);
+    }
+
+    return mergedRow;
```

Comment:
This logic is similar in every method, try to put it in a helper function and use it everywhere.